.github/workflows/run_tests.yml

```
@@ -96,7 +96,7 @@ jobs:
                      ghcr.io/${{ github.repository }}:${{ inputs.build-tag }} \
                      sh -c 'pip install -r requirements_test.txt && pytest -n4 --cov --cov-report=xml:/report/coverage.xml'
 
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v4
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
```

```suggestion
        env:
          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Let's perhaps try this way, as it stands in the docs: https://docs.codecov.com/docs/adding-the-codecov-token